### PR TITLE
[GLUTEN-3760][VL]fix unsupported column name corner case

### PR DESCRIPTION
**Pull request by zhli1142015**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#ISSUE-ID)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by zhli1142015** [comment]
@JkSelf  and @rui-mo, could you please help check this? Thanks.

File: gluten-core/src/main/scala/io/glutenproject/execution/FileSourceScanExecTransformer.scala [hunk 1]
```diff
@@ -108,8 +108,9 @@ class FileSourceScanExecTransformer(
   override def hashCode(): Int = super.hashCode()
 
   override protected def doValidateInternal(): ValidationResult = {
-    if (hasMetadataColumns) {
-      return ValidationResult.notOk(s"Unsupported metadataColumns scan in native.")
+    if (!unsupportedColumns.isEmpty) {
+      return ValidationResult.notOk(
+        s"Unsupported columns in native scan: ${unsupportedColumns.mkString(",")}")
```
Comment (JkSelf) [inline gluten-core/src/main/scala/io/glutenproject/execution/FileSourceScanExecTransformer.scala:113]: @zhli1142015 Can we modify the reason for unsupport to: 'Unsupported columns in native scan due to column name ambiguity in Velox. Please consider renaming the column.'? It is more readable.

**Review by JkSelf — COMMENTED** [review]
LGTM. Just one minor comment.

File: shims/spark32/src/main/scala/org/apache/spark/sql/execution/FileSourceScanExecShim.scala [hunk 2]
```diff
@@ -53,6 +53,20 @@ class FileSourceScanExecShim(
   // Note: "metrics" is made transient to avoid sending driver-side metrics to tasks.
   @transient override lazy val metrics: Map[String, SQLMetric] = Map()
 
+  val unsupportedColumns: Seq[String] = {
```
Comment (rui-mo) [inline shims/spark32/src/main/scala/org/apache/spark/sql/execution/FileSourceScanExecShim.scala:56]: FileSourceScan targets on datasource v1, do we have similar issue for datasource v2? If so, can this part of logic be put in `BasicScanExecTransformer`? Maybe by adding a function receives output as parameter.

File: shims/spark32/src/main/scala/org/apache/spark/sql/execution/FileSourceScanExecShim.scala [hunk 3]
```diff
@@ -53,6 +53,20 @@ class FileSourceScanExecShim(
...
+    // Below name has special meaning in Velox.
```
Comment (rui-mo) [inline shims/spark32/src/main/scala/org/apache/spark/sql/execution/FileSourceScanExecShim.scala:57]: What is the special meaning, can we add more details?

**Comment by zhli1142015** [comment]
Will submit another PR for this.

**Pull request closed without merging**